Assignment4/MyLibrary/MyLibrary.Tests/NumSetTests.cs

```diff
@@ -0,0 +1,239 @@
+            int[] array = new int[] { 1, 2, 3, 4, 5 };
+            NumSet numSet = new NumSet(array);
+
+            //Act
+            _ = new HashSet<int>(array);
+
+            //Assert
+            Assert.IsNotNull(numSet.MySet);
+        }
+
+        [TestMethod]
+        public void ToString_TakesHashSet_ReturnsStringOfHashSet()
+        {
+            //Assign
+            int[] array = new int[] { 1, 2, 3, 4, 5 };
+            NumSet numSet = new NumSet(array);
+
+            //Act
+            string arrayToString = numSet.ToString();
+
+            //Assert
+            Assert.AreEqual(arrayToString, "1, 2, 3, 4, 5");
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void ToString_TakesNullHashSet_ThrowsArgumentNullException()
+        {
+            //Assign
+            NumSet? numSet = new NumSet
+            {
+                MySet = null
+            };
+
+            //Act
+            _ = numSet.ToString();
+
+            //Assert
+        }
+
+        [TestMethod]
+        public void Equals_PassesDifferentNumSets_ReturnsFalse()
+        {
+            //Assign
+            int[] array1 = new int[] { 1, 2, 3, 4, 5 };
+            int[] array2 = new int[] { 6, 7, 8, 9, 10 };
+            NumSet numSet1 = new NumSet(array1);
+            NumSet numSet2 = new NumSet(array2);
+
+            //Act
+
+            //Assert
+            Assert.IsFalse(numSet1.Equals(numSet2));
+        }
+
+        [TestMethod]
+        public void Equals_PassesTheSameNumSet_ReturnsTrue()
+        {
+            //Assign
+            int[] array = new int[] { 1, 2, 3, 4, 5 };
+            NumSet numSet = new NumSet(array);
+
+            //Act
+
+            //Assert
+            Assert.IsTrue(numSet.Equals(numSet));
+        }
+
+        [TestMethod]
+        public void Equals_PassesNonNumSetObject_ReturnsFalse()
+        {
+            //Assign
+            int[] array = new int[] { 1, 2, 3, 4, 5 };
+            NumSet numSet = new NumSet(array);
+            object obj = new object();
+
+            //Act
+
+            //Assert
+            Assert.IsFalse(numSet.Equals(obj));
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void Equals_PassesNullArray_ThrowsArgumentNullException()
+        {
+            //Assign
+            int[] array1 = new int[] { 1, 2, 3, 4, 5 };
+            int[] array2 = new int[] { 6, 7, 8, 9, 10 };
+            NumSet numSet1 = new NumSet(array1);
+            NumSet numSet2 = new NumSet(array2);
+            numSet1.MySet = null;
+
+            //Act
+            _ = numSet1.Equals(numSet2);
+
+            //Assert
+        }
+
+        [TestMethod]
+        public void GetHashCode_TakesNumSet_ReturnsCorrectHashCode()
+        {
+            //Assign
+            int[] array = new int[] { 1, 2, 3, 4, 5 };
+            NumSet numSet1 = new NumSet(array);
+            NumSet numSet2 = numSet1;
```

Comment:
```suggestion
            int[] array = new int[] { 1, 2, 3, 4, 5 };
            NumSet numSet1 = new NumSet(array);
            NumSet numSet2 = new NumSet(array);
```
You should create two separate objects, rather than setting the second `NumSet` equal to the first one.